### PR TITLE
Add a method to extract identities from SAN extension

### PR DESCRIPTION
**Pull request by lookuptable**
This will be used by the grpc server for client certificate based
authN/authZ.

Also grouping all SAN-related operations into package `pkg/pki/san.go`

**Release note**:

```
NONE
```

### CODE REVIEW

File: pkg/pki/san.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+func ExtractSANExtension(cert *x509.Certificate) (*pkix.Extension, bool) {
+	for _, ext := range cert.Extensions {
+		if ext.Id.Equal(oidSubjectAlternativeName) {
+			// We don't need to examine other extensions anymore since a certificate
+			// must not include more than one instance of a particular extension. See
+			// https://tools.ietf.org/html/rfc5280#section-4.2.
+			return &ext, true
+		}
+	}
+	return nil, false
```
Comment (wattli) [inline pkg/pki/san.go]: I feel the bool is unnecessary, nil means false already, right?

Comment (lookuptable) [inline pkg/pki/san.go]: Yeah. we can check for nil as well.

**Comment by yutongz** [comment]
@wattli  Could you comment them in two different lines

**Comment by yutongz** [comment]
The pr got merged, yeah!!  Thank you guys for trusting the merge robot.
So basically, we you want to actually merge the pr, you need to comment "/lgtm" and "approve"
The reason why we need two level approvals is to bring more flexible. For example, you can ask Wencheng and Tao to review this at the same time. Wencheng can say "lgtm" after he is ok with it. But Tao may have some nit changes. I understand in many case, one pr only has one reviewer, especially for auth repo. So if you really don't like the two level approval, we can totally change to only "/lgtm" is required.
Feel free to bug me when you have problems with this robot.

**Pull request merged**